src/base/QXmppGlobal.h

```diff
@@ -47,6 +48,55 @@
 /// \since QXmpp 1.5
 ///
 namespace QXmpp {
+///
+/// This enum describes different end-to-end encryption methods. These can
+/// be used to mark a stanza as encrypted or decrypted with a specific algorithm
+/// (e.g., for \xep{0380, Explicit Message Encryption}).
+///
+/// \since QXmpp 1.5
+///
+enum Encryption {
+    /// No encryption
+    NoEncryption,
+    /// Unknown encryption
+    UnknownEncryption,
+    /// \xep{0364, Current Off-the-Record Messaging Usage}
+    Otr,
+#if QXMPP_DEPRECATED_SINCE(1, 5)
+    /// \xep{0364, Current Off-the-Record Messaging Usage}
+    /// \deprecated This enum is deprecated since QXmpp 1.5. Use
+    /// \c QXmpp::Otr instead.
+    OTR = Otr,
+#endif
```

Comment:
You can put all deprecated enumerations to the bottom, so you only need one `#if QXMPP_DEPRECATED_SINCE(1, 5)`.